### PR TITLE
DNS refactor: make DNS a first class intercept

### PR DESCRIPTION
**Pull request by ekoby**

### CODE REVIEW

File: lib/ziti-tunnel/ziti_tunnel.c [hunk 1]
```diff
@@ -283,23 +240,9 @@ address_t *parse_address(const char *hn_or_ip_or_cidr, dns_manager *dns) {
...
-            } else {
-                TNL_LOG(DEBUG, "intercept hostname %s is not an ip", addr->str);
-                if (ipaddr_aton(resolved_ip_str, &addr->ip) == 0) {
-                    TNL_LOG(ERR, "dns manager provided unparsable ip address '%s'", resolved_ip_str);
-                    free(addr);
-                    return NULL;
-                }
-            }
-        }
+        TNL_LOG(ERR, "hostnames are not supported");
```
Comment (scareything) [inline lib/ziti-tunnel/ziti_tunnel.c:243]: :(

File: lib/ziti-tunnel/ziti_tunnel.c [hunk 2]
```diff
@@ -271,7 +228,7 @@ void intercept_ctx_add_protocol(intercept_ctx_t *ctx, const char *protocol) {
     STAILQ_INSERT_TAIL(&ctx->protocols, proto, entries);
 }
 
-address_t *parse_address(const char *hn_or_ip_or_cidr, dns_manager *dns) {
+address_t *parse_address(const char *hn_or_ip_or_cidr) {
```
Comment (scareything) [inline lib/ziti-tunnel/ziti_tunnel.c:231]: Change the arg name to `ip_or_cidr`?

File: lib/ziti-tunnel/ziti_tunnel.c [hunk 3]
```diff
@@ -283,23 +240,9 @@ address_t *parse_address(const char *hn_or_ip_or_cidr, dns_manager *dns) {
     }
 
     if (ipaddr_aton(addr->str, &addr->ip) == 0) {
-        addr->is_hostname = true;
```
Comment (scareything) [inline lib/ziti-tunnel/ziti_tunnel.c:286]: is there any reason to keep `is_hostname` now?

File: lib/ziti-tunnel-cbs/ziti_dns.c [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+
+    ziti_dns.ip_pool.base = mask;
+    ziti_dns.ip_pool.counter = 10;
+    ziti_dns.ip_pool.counter_mask = ~( (uint32_t)-1 << (32 - (uint32_t)bits));
+}
+
+int ziti_dns_setup(tunneler_context tnlr, const char *dns_addr, const char *dns_cidr) {
+    seed_dns(dns_cidr);
+
+    intercept_ctx_t *dns_intercept = intercept_ctx_new(tnlr, "ziti:dns-resolver", &ziti_dns);
```
Comment (scareything) [inline lib/ziti-tunnel-cbs/ziti_dns.c:77]: cool!

File: lib/ziti-tunnel-cbs/ziti_dns.c [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+    intercept_ctx_add_address(dns_intercept, dns_addr);
+    intercept_ctx_add_port_range(dns_intercept, 53, 53);
+    intercept_ctx_add_protocol(dns_intercept, "udp");
```
Comment (scareything) [inline lib/ziti-tunnel-cbs/ziti_dns.c:80]: any reason not to add tcp also?

**Review by scareything — APPROVED** [review]

File: lib/ziti-tunnel/ziti_tunnel.c (hunk 3 above)
Comment (ekoby) [reply lib/ziti-tunnel/ziti_tunnel.c:286]: removing

File: lib/ziti-tunnel-cbs/ziti_dns.c (hunk 5 above)
Comment (ekoby) [reply lib/ziti-tunnel-cbs/ziti_dns.c:80]: may do it separately

**Pull request merged**